### PR TITLE
Upgrade to jakarta namespace

### PR DESCRIPTION
**Pull request by eocantu**
**Summary of the changes:**
* Updated to Jakarta EE 9 artifacts (jakarta namespace)
  * `jakarta.ws.rs:jakarta.ws.rs-api` -> `3.0.0`
  * `jakarta.annotation:jakarta.annotation-api` -> `2.0.0`
  * `jakarta.inject:jakarta.inject-api` -> `2.0.1`
  * `jakarta.servlet:jakarta.servlet-api` -> `5.0.0`
* New `jersey3-pac4j` module
* New `resteasy6-pac4j` module
* Removed `jersey2`, `jersey225`, `jersey228`, `resteasy3`, `resteasy5` modules since they aren't backwards compatible with JAX-RS 3

### CODE REVIEW

File: core/src/main/java/org/pac4j/jax/rs/filters/SecurityFilter.java [hunk 1]
```diff
@@ -56,7 +54,7 @@ protected void filter(JaxRsContext context) throws IOException {
         // For the later, we interpret the error and abort the request using jax-rs
         // abstractions
         buildLogic(config).perform(context, sessionStore, config, new SecurityGrantedAccessOutcome(), adapter(config),
-                clients, authorizers, matchers, multiProfile);
```
Comment (eocantu) [inline core/src/main/java/org/pac4j/jax/rs/filters/SecurityFilter.java:59]: I didn't see `multiProfile` as one of the parameters in [SecurityLogic interface](https://github.com/pac4j/pac4j/blob/pac4j-parent-5.7.1/pac4j-core/src/main/java/org/pac4j/core/engine/SecurityLogic.java#L31-L32) so I went ahead and deleted it. 

It seems to me this was missed in the upgrade to pac4j v5 last time. So I can make a separate PR for this if needed.

File: core/src/main/java/org/pac4j/jax/rs/filters/CallbackFilter.java [hunk 2]
```diff
@@ -26,10 +26,6 @@ public class CallbackFilter extends AbstractFilter {
 
     private String defaultUrl;
 
-    private Boolean saveInSession;
```
Comment (eocantu) [inline core/src/main/java/org/pac4j/jax/rs/filters/CallbackFilter.java:29]: Similarly, I didn't see `saveInSession` in [CallbackLogic](https://github.com/pac4j/pac4j/blob/pac4j-parent-5.7.1/pac4j-core/src/main/java/org/pac4j/core/engine/CallbackLogic.java#L29-L30)

File: core/pom.xml [hunk 3]
```diff
@@ -4,15 +4,15 @@
   <parent>
     <groupId>org.pac4j.jax-rs</groupId>
     <artifactId>parent</artifactId>
-    <version>5.0.1-SNAPSHOT</version>
+    <version>6.0.0-SNAPSHOT</version>
```
Comment (eocantu) [inline core/pom.xml:7]: I went ahead and set this to the next major version. It seemed it was going to be too messy trying to keep both javax and jakarta artifacts.

**Review by richeyh — APPROVED** [review]
Nice work!
 @leleuj can we get some eyes on this?

**Comment by leleuj** [comment]
This is excellent! Merging for early adopters to test...

**Pull request merged**